smrf/data/wrf.py

```diff
@@ -0,0 +1,188 @@
+                                         'longitude', 'elevation'))
+
+        metadata['latitude'] = mlat.flatten()
+        metadata['longitude'] = mlon.flatten()
+        metadata['elevation'] = mhgt.flatten()
+        metadata = metadata.apply(apply_utm,
+                                  args=(self.topo.zone_number,),
+                                  axis=1)
+
+        self.metadata = metadata
+
+    def load(self):
+        """
+        Load the data from a netcdf file. This was setup to work with a WRF
+        output file, i.e. wrf_out so it's going to look for the following
+        variables:
+        - Times
+        - XLAT
+        - XLONG
+        - HGT
+        - T2
+        - DWPT
+        - GLW
+        - RAINNC
+        - CLDFRA
+        - UGRD
+        - VGRD
+
+        Each cell will be identified by grid_IX_IY
+
+        """
+
+        self._logger.info('Reading data coming from WRF output: {}'.format(
+            self.config['wrf_file']
+        ))
+        self.file = nc.Dataset(self.config['wrf_file'])
+
+        self.get_metadata()
+
+        # GET THE TIMES
+        time = self.get_file_times()
+        time_ind = (time >= self.start_date) & (time <= self.end_date)
+        time = time[time_ind]
+
+        # GET THE DATA, ONE AT A TIME
+        for variable, wrf_variable in self.BASE_VARIABLES.items():
+
+            self._logger.debug(
+                'Loading variable {} from WRF field {}'.format(
+                    variable, wrf_variable))
+
+            df = pd.DataFrame(index=time, columns=self.primary_id)
+            for i in self.station_index:
+                g = metadata_name_from_index(i)
+                df[g] = self.file.variables[wrf_variable][time_ind, i[0], i[1]]
+
+            # Set variable and subset by start and end time
+            setattr(self, variable, df)
+
+        # post process the loaded data
+        self.air_temp = self.air_temp - 273.15
+
+        self._logger.debug('Calculating vapor_pressure')
+        vp = satvp(self.dew_point.values)
+        self.vapor_pressure = pd.DataFrame(
+            vp, index=time, columns=self.primary_id)
+
+        self._logger.debug('Loading cloud_factor')
+        self.cloud_factor = pd.DataFrame(index=time, columns=self.primary_id)
+        cf = 1 - np.mean(self.file.variables['CLDFRA'][time_ind, :], axis=1)
+        for i in self.station_index:
+            g = metadata_name_from_index(i)
+            v = cf[:, i[0], i[1]]
+            self.cloud_factor[g] = v
+
+        self._logger.debug('Loading wind_speed and wind_direction')
+        self.wind_speed = pd.DataFrame(index=time, columns=self.primary_id)
+        self.wind_direction = pd.DataFrame(index=time, columns=self.primary_id)
+        min_speed = 0.47
+
+        u10 = self.file.variables['UGRD'][time_ind, :]
+        v10 = self.file.variables['VGRD'][time_ind, :]
+
+        # calculate the wind speed
+        s = np.sqrt(u10**2 + v10**2)
+        s[s < min_speed] = min_speed
```

Comment:
Same as with loading HRRR data. Feels like this can be done with one call instead of two

Reply:
Same as above, we'll let the wind module handle this